qiskit/pulse/compiler/target.py

```diff
@@ -151,156 +163,94 @@ def __init__(
                 the Qiskit :class:`.MeasurementFrame` object index.
                 This frame must track the rotating frame
                 of the measurement stimulus signal.
-            qubit_ports: A dictioanry of hardware port identifier keyed on
-                the Qiskit :class:`~qiskit.pulse.model.Qubit` object index.
-                This port must be used to drive qubit regardless of frames.
-            coupler_ports: A dictionary of hardware port identifier keyed on
-                the Qiskit :class:`.Port` object index.
-                This port must be used to drive multi-qubit interactions.
-            mixed_frams: A dictionary of avilable mixed frame resources keyed on
-                the unique identifier of the port. Values are list of frame identifiers
-                available for this port to form a mixed frame.
-
+            tx_ports: A list of dictionary representing a spec of transmission ports. 
+                The spec dictionary must contain "qubits", "op_type", "num_frames",
```

Comment:
Maybe this should be a class/protocol? I don't want to make it hard to extend the structure but it would be good to capture required fields for developers to reference.

Reply:
Good suggestion. It's turned into dataclass.